packages/next/src/seo/route/site-metadata/validator.ts

```diff
@@ -0,0 +1,76 @@
+import { getCleanExcerpt } from '@/seo/utils';
+import type { SiteMetadata } from './types';
+import { RouteMetadataSchema } from './schema';
+import type { RouteValidator } from '../types';
+
+/**
+ * Validates the route metadata from the given data.
+ *
+ * @param data - The data to parse.
+ * @return The parsed route metadata.
+ */
+const validateRouteSiteMetadata: RouteValidator< SiteMetadata > = (
+	data: unknown
+) => {
+	const parsedData = RouteMetadataSchema.safeParse( data );
+	if ( ! parsedData.success || ! parsedData.data.nodeByUri ) {
+		return {};
+	}
+
+	const node = parsedData.data.nodeByUri;
+	const metadata: SiteMetadata = {};
+
+	switch ( node.__typename ) {
```

Comment:
1. NEVER use introspection for data, since it's disabled for public sites.
2. As noted on the queries, we should be able to build these non-deterministally, based on the available `node[ ${key} ]`, instead of needing to explicitly support a node.